pandas/core/common.py

```diff
@@ -236,7 +238,8 @@ def asarray_tuplesafe(values: Iterable, dtype: NpDtype | None = None) -> ArrayLi
     try:
         with warnings.catch_warnings():
             # Can remove warning filter once NumPy 1.24 is min version
-            warnings.simplefilter("ignore", np.VisibleDeprecationWarning)
+            if Version(np.__version__) < Version("1.24.0"):
```

Comment:
you can use
https://github.com/pandas-dev/pandas/blob/92d1d6a210f138bf368b189db2714839f5cf762a/pandas/compat/numpy/__init__.py#L9
instead here and below.

Reply:
Sure! I updated it to use these helpers.